components/Canvas.js

```diff
@@ -0,0 +1,197 @@
+                    contiguousLeft = true;
+                    stack.push({ x: operator.x - 1, y: operator.y });
+                }
+            } else {
+                contiguousLeft = false;
+            }
+
+            // Check right
+            if (
+                operator.x + 1 < width &&
+                colorMatch(getColorAtPixel(imageData, operator.x + 1, operator.y), baseColor)
+            ) {
+                if (!contiguousRight) {
+                    stack.push({ x: operator.x + 1, y: operator.y });
+                    contiguousRight = true;
+                }
+            } else {
+                contiguousRight = false;
+            }
+
+            operator.y++;
+            contiguousDown = colorMatch(getColorAtPixel(imageData, operator.x, operator.y), baseColor);
+        }
+    }
+};
+
+const Canvas = ({ canvasRef, addSnapShot, isFillMode, color, size, ...props }) => {
+    const [isPainting, setIsPainting] = React.useState(false);
+    const [mousePosition, setMousePosition] = React.useState(null);
+
+    const getCoordinates = (event) => {
+        const canvas = canvasRef.current;
+        return { x: event.pageX - canvas.offsetLeft, y: event.pageY - canvas.offsetTop };
+    };
+
+    const drawLine = (originalMousePosition, newMousePosition) => {
+        const context = canvasRef.current.getContext('2d');
+        context.strokeStyle = color;
+        context.lineJoin = 'round';
+        context.lineWidth = size;
+
+        context.beginPath();
+        context.moveTo(originalMousePosition.x, originalMousePosition.y);
+        context.lineTo(newMousePosition.x, newMousePosition.y);
+        context.closePath();
+
+        context.stroke();
+    };
+
+    const fill = ({ x, y }) => {
+        const canvas = canvasRef.current;
+        const context = canvas.getContext('2d');
+        const imageData = context.getImageData(0, 0, canvas.width, canvas.height);
+        floodFill(imageData, hexToRgba(color), x, y);
+        context.putImageData(imageData, 0, 0);
+    };
+
+    React.useEffect(() => {
+        const canvas = canvasRef.current;
+        const context = canvas.getContext('2d');
+        const imageData = context.getImageData(0, 0, canvas.width, canvas.height);
+        addSnapShot(imageData);
+    }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+    //Start drawing when the mouse is pressed.
+    React.useEffect(() => {
+        const startPaint = (event) => {
+            const coordinates = getCoordinates(event);
+            if (isFillMode) {
+                fill(coordinates);
+                return;
+            }
+
+            setMousePosition(coordinates);
+            setIsPainting(true);
+        };
+
+        const canvas = canvasRef.current;
+        canvas.addEventListener('mousedown', startPaint);
+        return () => {
+            canvas.removeEventListener('mousedown', startPaint);
+        };
+    });
+
+    //Draw the line on mouse move
+    React.useEffect(() => {
```

Comment:
I don't understant why these 3 effects don't have dependencies list ? 
Could it be called only at mount ?

Reply:
Hep, fix it :)